src/hera/events/service.py

```diff
@@ -50,7 +50,24 @@ def __init__(
         """Events service constructor."""
         self.host = cast(str, host or global_config.host)
         self.verify_ssl = verify_ssl if verify_ssl is not None else global_config.verify_ssl
-        self.token = token or global_config.token
+
+        # some users reported in https://github.com/argoproj-labs/hera/issues/1016 that it can be a bit awkward for
+        # Hera to assume a `Bearer` prefix on behalf of users. Some might pass it and some might not. Therefore, Hera
+        # only prefixes the token with `Bearer ` if it's not already specified and lets the uses specify it otherwise.
+        # Note that the `Bearer` token can be specified through the global configuration as well. In order to deliver
+        # a fix on Hera V5 without introducing breaking changes, we have to support both
+        global_config_token = global_config.token  # call only once because it can be a user specified function!
+        if token and token.lower().startswith("bearer"):
```

Comment:
I think a better logic would be to split this header on whitespace and if it has only one part, prefix it with Bearer otherwise allow users to specify the scheme.

Reply:
Can you say a bit more about why? I know this is relatively trivial but I chose to be explicit about the "bearer" part so it's clear what Hera's looking for. Doing it on length seems a bit more implicit. While it can surely be documented it's not as "self describing". Open to switching it, of course, just want to know more!

Reply:
I was doing a bit of research (new to me as well) - https://stackoverflow.com/a/77426797 - basically `Bearer` is an authentication scheme but it isn't the _only_ option

Reply:
Ah, got it